File: app/models/company.py
import enum
from sqlalchemy.orm import relationship
from sqlalchemy import Column, Integer, String, Enum, Text
from app.db.database import Base
from app.models.base import TimestampMixin

class IndustrySector(str, enum.Enum):
    """
    Using an Enum (not free-text strings) means only valid sectors
    can be stored. Bad data is rejected at the DB level, not just
    in Python. This is data integrity.
    """

    ENERGY = "energy"
    MANUFACTURING = "manufacturing"
    TRANSPORT = "transport"
    FINANCE = "finance"
    TECHNOLOGY = "technology"
    RETAIL = "retail"
    HEALTHCARE = "healthcare"
    OTHER = "other"

class Company(TimestampMixin, Base):
    """
    __tablename__ tells SQLAlchemy what the actual table is called in PostgreSQL.
    Every column is defined as a class attribute with its type and constraints.
    """

    __tablename__ = "companies"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False, unique=True, index=True)
    sector = Column(Enum(IndustrySector), nullable=False)
    country = Column(String(100), nullable=False, default="France")
    description = Column(Text, nullable=True)
    employee_count = Column(Integer, nullable=True)
    annual_revenue_eur = Column(Integer, nullable=True)

    emission_records = relationship(
        "EmissionRecord",
        back_populates="company",
        cascade = "all, delete-orphan"
    )

    def __repr__(self):
        return f"<Company id = {self.id} name = {self.name}>"
    

File: frontend/src/types/company.ts
export type IndustrySector =
  | 'energy' | 'manufacturing' | 'transport'
  | 'finance' | 'technology' | 'retail'
  | 'healthcare' | 'other'

export interface Company {
  id: number
  name: string
  sector: IndustrySector
  country: string
  description: string | null
  employee_count: number | null
  annual_revenue_eur: number | null
  created_at: string
  updated_at: string
}

export interface CompanyListResponse {
  items: Company[]
  total: number
  page: number
  page_size: number
}

export interface CompanyCreate {
  name: string
  sector: IndustrySector
  country: string
  description?: string
  employee_count?: number
  annual_revenue_eur?: number
}
